syntax = "proto3";

package recommendation;

service RecommendationService {
  rpc GetProductRecommendations (ProductRecommendationRequest) returns (ProductRecommendationResponse) {}
  rpc GetUserRecommendations (UserRecommendationRequest) returns (ProductRecommendationResponse) {}
}

message ProductRecommendationRequest {
  string product_id = 1;
  int32 max_results = 2;
}

message UserRecommendationRequest {
  string user_id = 1;
  int32 max_results = 2;
}

message Product {
  string id = 1;
  string name = 2;
  double score = 3;
}

message ProductRecommendationResponse {
  repeated Product products = 1;
} 